dspace-api/src/main/java/org/dspace/authenticate/OidcAuthentication.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.authenticate;
+
+import java.sql.SQLException;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.Group;
+import org.dspace.kernel.ServiceManager;
+import org.dspace.utils.DSpace;
+
+/**
+ * Implementation of {@link AuthenticationMethod} that delegate all the method
+ * invocations to the bean of class {@link OrcidAuthenticationBean}.
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ *
+ */
+public class OidcAuthentication implements AuthenticationMethod {
```

Comment:
I have to admit, this design of having an AuthenticationMethod which wraps a separate Bean which is also an AuthenticationMethod is very odd.  Could @LucaGiamminonni or someone explain why we'd want to do this?  Why do we need both an `OidcAuthentication` and `OidcAuthenticationBean`, both of which implement `AuthenticationMethod`?  Shouldn't we just use one like we do with all other Auth classes?

Reply:
Hi @tdonohue, The intent of this class is to be able to use the OidcAuthenticationBean class as a spring bean, with all the advantages that it entails: for example, I can take advantage of dependency injection instead of having to do the lookup to derive the dependencies and this also simplifies the test. The very odd thing for me is not the OidcAuthenticationBean class which is structured to be a classic spring bean but is that the others AuthenticationMethod are not beans of Spring. If someday we make all the other Spring AuthenticationMethod become beans (which I hope) the OidcAuthenticationBean class does not need to change.